### PR TITLE
TaskSorting

### PR DESCRIPTION
**Pull request by StanislavOlijnik**

### CODE REVIEW

File: components/TodoList.js [hunk 1]
```diff
@@ -36,20 +38,26 @@ function TodoList() {
     const updatedTodos = todos.map((todo) =>
       todo.id === id ? { ...todo, text: editedTodoText } : todo
     );
-  
-  
-    if (editedTodoText.trim() === "") {
-      console.log();
-      
+
+    if (editedTodoText.trim() === '') {
```
Comment (DorisW3) [inline components/TodoList.js:42]: I  don't understand line 42/43. What function do they have?

File: components/TodoList.js [hunk 2]
```diff
@@ -58,35 +66,97 @@ function TodoList() {
...
+        <option value="low">Low Priority</option>
+      </select>
       <button onClick={addNewTodo}>Add</button>
-      <ul>
-        {todos.map((todo) => {
-          if (editTodoId === todo.id) {
-            return (
+
+      <div style={{ display: 'flex' }}>
+        <div style={{ flex: 1 }}>
```
Comment (DorisW3) [inline components/TodoList.js:77]: You use inline style here. Better use styled-components in general.

File: components/TodoList.js [hunk 3]
```diff
@@ -58,35 +66,97 @@ function TodoList() {
...
+                    {todo.text}
+                    <button onClick={() => removeTodo(todo.id)}>Delete</button>
+                    <button onClick={() => startEditing(todo.id, todo.text)}>Edit</button>
+                  </>
+                )}
+              </li>
+            ))}
+          </ul>
+        </div>
+        <div style={{ flex: 1 }}>
```
Comment (DorisW3) [inline components/TodoList.js:103]: the same as the comment for line 76/77

File: components/TodoList.js [hunk 4]
```diff
@@ -58,35 +66,97 @@ function TodoList() {
...
+                  </>
+                )}
               </li>
-            );
-          } else {
-            return (
+            ))}
+          </ul>
+        </div>
+        <div style={{ flex: 1 }}>
```
Comment (DorisW3) [inline components/TodoList.js:129]: look above

**Review by DorisW3 — CHANGES_REQUESTED** [review]
There are only a few changes you should make. ;-)

File: components/TodoList.js (hunk 2 above)
Comment (StanislavOlijnik) [reply components/TodoList.js]: Thank you, it's a very good idea, I've now created a new folder for the styled components

File: components/TodoList.js (hunk 3 above)
Comment (StanislavOlijnik) [reply components/TodoList.js]: Thank you, it's a very good idea, I've now created a new folder for the styled components

File: components/TodoList.js [hunk 5]
```diff
@@ -1,89 +1,135 @@
...
-    if (newTodo.trim() !== '') {
-      const todoItem = {
-        id: Math.random().toString(32).substring(2),
-        text: newTodo,
-      };
-      setTodos([...todos, todoItem]);
-      setNewTodo('');
-    }
-  }
+  const handleNewTodoChange = (e) => {
```
Comment (F-Kirchhoff) [inline components/TodoList.js:36]: please use event instead of "e"

File: components/TodoList.js [hunk 6]
```diff
@@ -1,89 +1,135 @@
...
+    setNewTodo(e.target.value);
+  };
 
-  function removeTodo(id) {
-    const newTodos = todos.filter((todo) => todo.id !== id);
-    setTodos(newTodos);
-  }
+  const handlePriorityChange = (e) => {
```
Comment (F-Kirchhoff) [inline components/TodoList.js:40]: same here

File: components/TodoList.js [hunk 7]
```diff
@@ -1,89 +1,135 @@
...
+  const handleRemoveTodo = (id) => {
+    const newTodos = todos.filter((todo) => todo.id !== id);
+    setTodos(newTodos);
+  };
+
+  const priorityGroups = {
+    high: todos.filter((todo) => todo.priority === 'high'),
+    medium: todos.filter((todo) => todo.priority === 'medium'),
+    low: todos.filter((todo) => todo.priority === 'low'),
+  };
```
Comment (F-Kirchhoff) [inline components/TodoList.js:89]: cool!

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Stanislav,
this looks very good, please adjust only the naming of some variables.

File: components/TodoList.js [hunk 8]
```diff
@@ -1,89 +1,135 @@
...
+`;
+
+const TodoItem = styled.li`
+  display: flex;
+  align-items: center;
+  margin-bottom: 4px;
+`;
+
+const TodoText = styled.span`
+  flex: 1;
```
[the remark this answers is not in the document]
Comment (StanislavOlijnik) [reply components/TodoList.js]: yes it works very well without it, thank you very much.

File: components/TodoList.js (hunk 5 above)
Comment (StanislavOlijnik) [reply components/TodoList.js]: yes is better, I changed it.

File: components/TodoList.js (hunk 6 above)
Comment (StanislavOlijnik) [reply components/TodoList.js]: yes is better, I changed it

**Review by F-Kirchhoff — APPROVED** [review]
LGTM 🦊

**Pull request merged**